bots/push-rewrite

```diff
@@ -30,6 +30,20 @@ def execute(*args):
 def git(*args):
     return execute("git", *args).strip()
 
+def find_pr_with_sha(api, sha):
+    pulls = api.pulls()
+    for pull in pulls:
+        if pull["head"]["sha"] == sha:
+            return pull
+    sys.stderr.write("Could not find pull with revision {0}.\n".format(sha))
+    return 1
+
+def labels_of_pr(pull):
+    if "labels" not in pull:
+        pull["labels"] = api.get("issues/{0}/labels".format(number))
+    return list(map(lambda label: label["name"], pull["labels"]))
```

Comment:
This duplicates code  from bots/tests-scan. Can this please move into bots/task/github.py?